apps/documents/views.py

```diff
@@ -0,0 +1,248 @@
+from django.urls import reverse
+from django.views.decorators.http import require_POST
+from django.views.generic import DetailView, ListView, TemplateView
+
+from apps.documents.models import Repository, RepositoryType
+from apps.files.models import MAX_SUMMARY_LENGTH, File
+from apps.teams.decorators import login_and_team_required
+from apps.teams.mixins import LoginAndTeamRequiredMixin
+
+
+class RepositoryHome(LoginAndTeamRequiredMixin, TemplateView):
+    template_name = "documents/repositories.html"
+
+    def get_context_data(self, team_slug: str, tab_name: str, **kwargs):
+        context = {
+            "active_tab": "manage_files",
+            "title": "Manage Files",
+            "tab_name": tab_name,
+            "files_list_url": reverse("documents:files_list", kwargs={"team_slug": team_slug}),
+            "upload_files_url": reverse("documents:upload_files", kwargs={"team_slug": team_slug}),
+            "collections_list_url": reverse("documents:collections_list", kwargs={"team_slug": team_slug}),
+            "new_collection_url": reverse("documents:new_collection", kwargs={"team_slug": team_slug}),
+            "files_count": File.objects.filter(team__slug=team_slug, external_id="").count(),
+            "max_summary_length": MAX_SUMMARY_LENGTH,
+            "collections_count": Repository.objects.filter(
+                team__slug=team_slug, type=RepositoryType.COLLECTION
+            ).count(),
+        }
+
+        if tab_name == "files":
+            context["collections"] = Repository.objects.filter(
+                team__slug=team_slug, type=RepositoryType.COLLECTION
+            ).all()
+
+        return context
+
+
+class BaseObjectListView(ListView, PermissionRequiredMixin):
+    details_url_name: str
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context["details_url_name"] = self.details_url_name
+        context["tab_name"] = self.tab_name
+        return context
+
+
+class FileListView(LoginAndTeamRequiredMixin, BaseObjectListView):
+    template_name = "documents/shared/list.html"
+    model = File
+    paginate_by = 10
+    details_url_name = "documents:file_details"
+    tab_name = "files"
+    permission_required = "files.view_file"
+
+    def get_queryset(self):
+        queryset = (
+            super().get_queryset().filter(team__slug=self.kwargs["team_slug"], external_id="").order_by("-created_at")
+        )
+        search = self.request.GET.get("search")
+        if search:
+            # TODO: Expand to search summary as well
+            name_similarity = TrigramSimilarity("name", search)
+            queryset = (
+                queryset.annotate(similarity=name_similarity).filter(Q(similarity__gt=0.2)).order_by("-similarity")
+            )
+        return queryset
+
+
+class BaseDetailsView(DetailView, PermissionRequiredMixin):
+    pass
+
+
+class FileDetails(LoginAndTeamRequiredMixin, BaseDetailsView):
```

Comment:
nit: could add `LoginAndTeamRequiredMixin` to the `BaseDetailsView` class

Reply:
https://github.com/dimagi/open-chat-studio/pull/1259/commits/bccd4ccc1889c988b6f6c4775b61d5a7854be9d6